pandas/core/generic.py

```diff
@@ -5906,6 +5908,129 @@ def infer_objects(self: FrameOrSeries) -> FrameOrSeries:
             )
         ).__finalize__(self)
 
+    # ----------------------------------------------------------------------
+    # Convert to types that support pd.NA
+
+    def _as_nullable_dtype(self: ABCSeries) -> ABCSeries:
+        """
+        Handle one Series
+
+        Rules:
+        If an object, see if we can infer string, boolean or integer, otherwise leave
+        alone
+        If an integer and not an extension type, convert to the Int64/Int32 type
+        (platform dependent)
+        If numeric, see if we can infer integer, otherwise try to use astype() to make
+        it integer.
+
+        """
+        dtype = self.dtype
+        new_dtype = dtype
+        changeit = False
+        constructit = True
+        result = self
+        target_int_dtype = "Int64"
+
+        if is_object_dtype(dtype):
+            new_dtype = lib.infer_dtype(self)
+            if new_dtype not in {"string", "boolean", "integer"}:
```

Comment:
infer_dtype can also return `mixed-integer`, which should perhaps be handled here.

Reply:
@TomAugspurger But `mixed-integer` would only happen if you had integers and other objects, in which case we can't make it a pure integer extension type. 
```python
In [1]: import pandas as pd

In [2]: import numpy as np

In [3]: from pandas._libs.lib import infer_dtype as idt

In [4]: idt(pd.Series([1,2,3], dtype=np.dtype('O')), skipna=True)
Out[4]: 'integer'

In [5]: idt(pd.Series([1,2,3.5], dtype=np.dtype('O')), skipna=True)
Out[5]: 'mixed-integer-float'

In [6]: idt(pd.Series([1,2,'s'], dtype=np.dtype('O')), skipna=True)
Out[6]: 'mixed-integer'
```

Reply:
Ah, forgot we had `skipna=True` now, so you are correct. Unless we want to handle the following,  which maybe we don't

```python
In [13]: pd._libs.lib.infer_dtype(np.array([1, 1.0], dtype=object))
Out[13]: 'mixed-integer-float'
```

Reply:
If it is mixed integer, I could do the same thing we do with float by trying to force the conversion. So then the results of ` pd.Series([1, 1.0], dtype=object).as_nullable_dtypes()` and `pd.Series([1, 1.0], dtype=np.dtype('int64')).as_nullable_dtypes()` would be the same.  I'll add that.

Reply:
done